components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -229,10 +230,16 @@ protected String getUserInfoEndpoint(OAuthClientResponse token, Map<String, Stri
 
             for (Map.Entry<String, Object> data : jsonObject.entrySet()) {
                 String key = data.getKey();
-                Object value = data.getValue();
+                Object valueObject = data.getValue();
 
-                if (value != null) {
-                    claims.put(ClaimMapping.build(key, key, null, false), value.toString());
+                if (valueObject != null) {
+                    String value;
+                    if (valueObject instanceof Object[]) {
+                        value = StringUtils.join((Object[]) valueObject, getMultiAttributeSeparator());
```

Comment:
Can't we use the util method from FrameworkUtils[1] here?

https://github.com/wso2/carbon-identity-framework/blob/d97cc31309ee6233bf151ca34c51e2dabfec971f/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java#L1631

Reply:
Resolved it by upgrading identity-framework version to 5.7.5